google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1321,6 +1339,19 @@ protected Builder setUseVirtualThreads(boolean useVirtualThreads) {
       return this;
     }
 
+    /**
+     * Sets whether to enable extended OpenTelemetry tracing. Enabling this option will add the
+     * following additional attributes to the traces that are generated by the client:
+     *
+     * <ul>
+     *   <li>db.statement: Contains the SQL statement that is being executed.
+     * </ul>
+     */
+    public Builder setEnableExtendedTracing(boolean enableExtendedTracing) {
```

Comment:
I would very much prefer to keep options as much as possible as s instance variables, and not static. Static options makes it hard to predict what happens if you change the value halfway through the lifetime of the application. Does it affect existing clients? Or only new ones?
Static options are also very hard to use in combination with the JDBC driver, as JDBC assumes that all configuration goes through the JDBC connection URL (so basically a string). If one JDBC connection enables extended tracing, it would enable it for all connections (where 'all' could mean 'all existing and future ones' or 'all future ones').

Setting an environment variable does indeed set it for all clients, but it is something that (in Java) can only be set at startup. Changing an environment variable after an application has started is not possible, which prevents the previously mentioned problem around 'what happens to existing clients if you change the value during the lifetime of the application?'.

Reply:
Makes sense, we faced enough problems earlier with static options. Let move forward with this instance option